tools/swift-stdlib-tool/swift-stdlib-tool.cpp

```diff
@@ -670,583 +643,576 @@ -(NSString *)sst_command {
+    if (Verbose > 1) {
+        arguments[3] = "--log";
+        argsCount += 1;
+    }
 
-        task.arguments = arguments;
-        
-        NSPipe *outPipe = [NSPipe pipe];
-        NSPipe *errPipe = [NSPipe pipe];
-        task.standardOutput = outPipe;
-        task.standardError = errPipe;
-        
-        log_v("  %s", task.sst_command.fileSystemRepresentation);
-        [task launch];
-
-        // Read stdout and stderr in parallel, then wait for the task 
-        // to exit. Anything else risks deadlock if the task fills 
-        // one of the output buffers.
-
-        __block NSData *stdErrData = nil;
-        dispatch_semaphore_t gotStdErr = dispatch_semaphore_create(0);
-        dispatch_queue_t concurrentQueue = 
-            dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0);
-        dispatch_async(concurrentQueue, ^{
-            stdErrData = 
-                [[errPipe.fileHandleForReading readDataToEndOfFile] retain];
-            dispatch_semaphore_signal(gotStdErr);
-        });
+    for (auto const &string : commandAndArguments) {
+        arguments[argsCount++] = string.c_str();
+    }
+    arguments[argsCount] = NULL;
 
-        NSData *stdOutData = [outPipe.fileHandleForReading readDataToEndOfFile];
+    int outPipe[2];
+    int errPipe[2];
 
-        dispatch_semaphore_wait(gotStdErr, DISPATCH_TIME_FOREVER);
-        dispatch_release(gotStdErr);
-        [stdErrData autorelease];
+    pipe(outPipe);
+    pipe(errPipe);
+    
+    log_v("  %s", arguments[0]);
 
-        [task waitUntilExit];
-        
-        // Task is finished and we have its stdout and stderr output.
+    int childPid = fork();
 
-        // Print its stdout and stderr if it failed or we are verbose.
-        // (Print nothing by default because codesign is noisy.)
+    if (childPid == 0) {
+        dup2(outPipe[1], STDOUT_FILENO);
+        dup2(errPipe[1], STDERR_FILENO);
 
-        int err = task.terminationStatus;
-        if (err  ||  Verbose > 1) {
-            fwrite(stdErrData.bytes, stdErrData.length, 1, stdout);
-            fwrite(stdOutData.bytes, stdOutData.length, 1, stdout);
-        }
-        if (block) { block(stdOutData, stdErrData, err); }
-        return err;
+        execv(launchPath, (char *const *)arguments);
+    }
+    
+    // Read stdout and stderr in parallel, then wait for the task 
+    // to exit. Anything else risks deadlock if the task fills 
+    // one of the output buffers.
+
+    int errPipeReadFd = errPipe[1];
+    __block std::vector<uint8_t> stdErrData;
+    dispatch_semaphore_t gotStdErr = dispatch_semaphore_create(0);
+    dispatch_queue_t concurrentQueue = 
+        dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0);
+    dispatch_async(concurrentQueue, ^{
+        stdErrData = readToEOF(errPipeReadFd);
+        dispatch_semaphore_signal(gotStdErr);
+    });
+
+    auto const stdOutData = readToEOF(outPipe[0]);
+
+    dispatch_semaphore_wait(gotStdErr, DISPATCH_TIME_FOREVER);
```

Comment:
Waiting forever doesn't seem like the right default. But it's also not your bug so we can deal with this later.